test/scholar/decomposition/pca_test.exs

```diff
@@ -4,25 +4,7 @@ defmodule Scholar.Decomposition.PCATest do
   @x Nx.tensor([[-1, -1], [-2, -1], [-3, -2], [1, 1], [2, 1], [3, 2]])
   @x2 Nx.tensor([[1, 4], [54, 6], [26, 7]])
   @x3 Nx.tensor([[-1, -1, 3], [-2, -1, 2], [-3, -2, 1], [3, 1, 1], [21, 2, 1], [5, 3, 2]])
-
-  def assert_all_close(left, right, opts \\ []) do
```

Comment:
You should keep `assert_all_close` and also move it to a new `ExUnit.CaseTemplate` which is to be used anywhere it's needed.